manifests/limits.pp

```diff
@@ -28,27 +30,40 @@
   validate_bool($purge_limits_d_dir_real)
 
   include pam
-
-  # ensure target exists
-  common::mkdir_p { $limits_d_dir: }
-
-  file { 'limits_d':
-    ensure  => directory,
-    path    => $limits_d_dir,
-    owner   => 'root',
-    group   => 'root',
-    mode    => $limits_d_dir_mode,
-    purge   => $purge_limits_d_dir_real,
-    recurse => $purge_limits_d_dir_real,
-    require => [ Package[$pam::my_package_name],
-                Common::Mkdir_p[$limits_d_dir],
-                ],
+  if $config_file_contents == undef and $config_file_source == undef {
+    $content = template('pam/limits.conf.erb')
+  } else {
+    # config_file_contents takes priority over config_file_source
+    if $config_file_contents == undef {
+      $content = undef
+      $config_file_source_real = $config_file_source
+    } else {
+      $config_file_source_real = undef
+      validate_array($config_file_contents)
+      $content = template('pam/limits.conf.erb')
+    }
+  }
+  if $::osfamily == 'Suse' and $::lsbmajdistrelease == '10'  {
+  } else {
+    common::mkdir_p { $limits_d_dir: }
+    file { 'limits_d':
+      ensure  => directory,
+      path    => $limits_d_dir,
+      owner   => 'root',
+      group   => 'root',
+      mode    => $limits_d_dir_mode,
+      purge   => $purge_limits_d_dir_real,
+      recurse => $purge_limits_d_dir_real,
+      require => [ Package[$pam::my_package_name],
+                  Common::Mkdir_p[$limits_d_dir],
+                  ],
+    }
   }
-
   file { 'limits_conf':
     ensure  => file,
     path    => $config_file,
-    source  => 'puppet:///modules/pam/limits.conf',
```

Comment:
we can change source to content and move the file from files/ to templates/ 

the outcome is the same, so this is OK. Just have to update spec tests